### ISSUE TITLE
Media query with \0 causing parsing issues with node-sass and LESS

### ISSUE THREAD

**Issue by jraadt** [angular/material#6304]
Because of the https://github.com/angular/material/commit/f3761781f74065c3d6cba8029a3ba420aab957ff commit, which includes a \0 after the media definition, applications using LESS and importing the generated CSS file with the less option to have access to classes and such within LESS no longer work.  LESS doesn't support the \0 media query hack.  Import example:
```
@import (less) 'angular-material/angular-material.css';
```
When trying to compile LESS you get the following error:
```
media definitions require block statements after any features in file angular-material\angular-material.css line no. 654
```

**Comment by barryvdh** [comment]
Also having this issue. Had to split up my vendor + app styling, but not ideal.

**Comment by larsRedia** [comment]
I think I have a similar problem using SASS:

    Error: Invalid CSS after "...a screen\\0  and": expected "{", was "-width: 599px) ..."
        on line 1286 of stdin
        >> @media screen\0  and (max-width: 599px) {
       -------^

**Comment by enghen10** [comment]
@jraadt I also ran into the same with less, and I see that you found the same feature/bug in the less repo as I did. https://github.com/less/less.js/issues/1538

Seems like this could be made supported with less anyway by a small modification to that media query?

**Comment by kouak** [comment]
Same issue using [node-sass](https://www.npmjs.com/package/node-sass) (which uses libsass under the hood).

**Comment by johnsheldon** [comment]
I can confirm the issue in node-sass as well. It's breaking everything since the angular-material.css module is not building when we deploy to our server.

**Comment by idoo** [comment]
same problem, when i try deploy to heroku

**Comment by Sayene** [comment]
+1

**Comment by stalinb87** [comment]
+1

**Comment by a-boertien** [comment]
+1

**Comment by simonh1000** [comment]
+1

**Comment by jcmccormick** [comment]
+1

**Comment by alextreppass** [comment]
Same issue with angular-material installed over bower. It (libsass?) chokes on the `\0` IE hacks:
```
>> Invalid CSS after "...ia screen\\0 and": expected "{", was "-width:599px){...."
>>   Line 6  Column 25446  bower_components/angular-material/angular-material.min.css
```

**Comment by ddjonline** [comment]
+1  angular-material 1.0.1 with node-sass ^3.3.3

**Comment by Egidius** [comment]
+1

**Comment by breity** [comment]
+1 me too with angular-material 1.0.1 and node-sass cli

**Comment by lasernite** [comment]
Was there a work around for this?

angular-material 1.0.1-master-d1b7f79

`Sass::SyntaxError: Invalid CSS after "...a screen\0 and ": expected "{", was "(max-width: 599..."`

**Comment by jcmccormick** [comment]
I, too, would love a work around for this, especially if this particular issue isn't going to be fixed until 1.1.0, which looks like it may be the case, looking at the backlog and milestones.

Currently, I'm stuck using 0.9.8, which cuts out a hefty chunk of functionality.

Using Ruby on Rails with the sass-rails gem.

**Comment by prijindal** [comment]
+1 for me too

**Comment by prijindal** [comment]
I was having the same problem while using gulp

What i did was before feeding it into the sass compiler, I replaced all '\\0' with a space ' '

```
  return gulp.src(assets.vendorCss.main)
    .pipe(replace('screen\\0','screen'))
    .pipe(concat(assets.vendorCss.output.filename))
    .pipe(sass({outputStyle: 'compressed'}))
    .pipe(gulp.dest(assets.vendorCss.output.path))
```

And now it works greatly
Note: I am using gulp-sass for compiling, i am not sure how gulp-ruby-sass handles it but i think this workaround should work for them

**Comment by alanwilhelm** [comment]
It was failing for me in rails using sass-rails. Sprockets doesn't make it easy to pipe it through a string replace function so. Changing line 666 of `angular-material.scss` from
` @media screen {\0`
to
` @media screen {` 

was all that was required for me to get it to compile.

**Comment by kavare** [comment]
@prijindal this way works for me! 
I am now with `angular-marterial 1.0.1` and `node-saas` in my gulp building process.

Just to add a point: `replace` is not a default function in gulp so you have to **install [gulp-replace](https://www.npmjs.com/package/gulp-replace) in advance.**

**Comment by newaeonweb** [comment]
@prijindal, @kavare : the workaround works fine for me, thanks

**Comment by ruischnaider** [comment]
+1

**Comment by tenshiAMD** [comment]
+1

**Comment by xiazek** [comment]
+1

**Comment by kmgilbert100** [comment]
+1

**Comment by AurelieV** [comment]
+1

**Comment by rafaelgaspar** [comment]
+1

**Comment by cerebroso** [comment]
+1

**Comment by andre-schilipack** [comment]
+1

**Comment by Fibrasek** [comment]
+1

**Comment by rafaelgaspar** [comment]
This has been fixed for sass with https://github.com/sass/sass/issues/1932 for their version 3.4.21.

And I think this issue is more on the renderer side than Angular Material, so I think they are the one that need to fix this.

If there isn't an issue on gulp project someone should probably make one.

**Comment by jwmay2012** [comment]
+1

**Comment by hinok** [comment]
Just for now...

```
# OS X
sed -i '' 's|screen\\0|screen|g' dist/modules/js/**/*.css

# Linux
# sed -i 's|screen\\0|screen|g' dist/modules/js/**/*.css
```

Hope it will help someone.

**Comment by jwmay2012** [comment]
What I ended up doing. 

Stopped including angular-material in my css.scss file then added 
```
<link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/angular_material/1.0.5/angular-material.min.css">
```
to my html.

**Comment by futurechan** [comment]
+1

**Comment by xiazek** [comment]
Temporary fix for rails, which worked for us : 

in `application.scss`
```javascript
@import "angular_material_fixed";
```

in `/app/assets/stylesheets/_angular_material_fixed.scss.erb`
```
<%= 
  # Workaround for this issue: https://github.com/angular/material/issues/6304
  # TODO: remove this once angular-material 1.0.6 is released
  original = File.read(File.expand_path('../../../../../vendor/assets/javascripts/bower_components/angular-material/angular-material.scss', __FILE__))
  original.gsub('screen\0', 'screen ')
%>
```

fingers crossed for a 1.0.6 release

**Comment by alexilyaev** [comment]
@ThomasBurleson I'm still getting an error using Sass with Webpack:
```
Module build failed:
@media screen\0  and (max-width: 599px) {
      ^
      Invalid CSS after "...a screen\\0  and": expected "{", was "-width: 599px) ..."
      in /Users/alex/ ... /node_modules/angular-material/angular-material.css (line 6569, column 8)
 @ ./~/angular-material/angular-material.css 4:14-230 13:2-17:4 14:20-236
```

Versions:
- `"angular-material": "1.0.6",`
- `"node-sass": "3.4.2",`
- `"sass-loader": "3.1.2",`

**Comment by andre-schilipack** [comment]
+1

**Comment by stefanahman** [comment]
Yep, still problematic: `Sass::SyntaxError: Invalid CSS after "...a screen\0 and ": expected "{", was "(max-width: 599..."`

**Comment by shershen08** [comment]
+1

**Comment by vsych** [comment]
+1

**Comment by DavidCheas** [comment]
+1

**Comment by alex-alexandrescu** [comment]
Fix i used:
`@media #{"screen\\0"}
  {}`

**Comment by stefanahman** [comment]
@ThomasBurleson Why is this still closed?

**Comment by stefanahman** [comment]
Could this be an issue? https://github.com/angular/material/commit/eaf0fe5193d862120eee673de0861632d076d4a7#diff-c85cbcb498085b434972d3ed6403735dR23

**Comment by ThomasBurleson** [comment]
@stefanahman, @alexalexandresq - thx for the information. Reopened.

**Comment by moeabdol** [comment]
I'm having the same problem. I ended up using the CDN version instead.

**Comment by ThomasBurleson** [comment]
* the `**ie_fixes.css**` should not be included in the Sass compile process. It is a standalone, post-compile resource.
* `[bower-material/modules/layouts/angular-material.layouts.ie_fixes.css](https://github.com/angular/bower-material/blob/master/modules/layouts/angular-material.layouts.ie_fixes.css)` contains valid `@media screen\0`.

**Comment by ThomasBurleson** [comment]
@gligoran - that should be removed also. Thank you for the alert.

**Comment by tenshiAMD** [comment]
This issue appeared again in the v1.0.8 in production deployment.

**Comment by mftaha** [comment]
+1

**Comment by ThomasBurleson** [comment]
Please Upgrade to at least 1.0.9 or (preferrably) v1.1.0-rc.5.

**Comment by ThomasBurleson** [comment]
@gligoran - Thx for the alert; not sure how we missed that. Updated in SHA cb59b08.

**Comment by twizzzlers** [comment]
+1

I'm in 1.0.9 and still facing this issue when trying to deploy on production, which work arround is more recommended?

**Comment by ThomasBurleson** [comment]
@twizzzlers - upgrade to v1.1.0-rc.5... which has so many more fixes and improvments than 1.0.9.

**Comment by jexneedls** [comment]
+1

Hi @ThomasBurleson, sorry to tell you this but i'm still getting the issue for both 1.0.9 and v1.1.0-rc.5.

**Comment by ThomasBurleson** [comment]
@jexneedls - how are you getting this with the v1.1.0-rc.5 or later ?

**Comment by vucalur** [comment]
fix in `v1.1.0-rc.5` works for me
